### PR TITLE
Use ccache to speed up rebuilds

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: test/python_tests.txt [hunk 1]
```diff
@@ -90,18 +85,18 @@ test_cmd_line_script  subprocess
 test_code
 test_code_module
 test_codeccallbacks
-test_codecencodings_cn  unknown encoding
-test_codecencodings_hk  unknown encoding
-test_codecencodings_iso2022 unknown encoding
-test_codecencodings_jp      unknown encoding
-test_codecencodings_kr  unknown encoding
-test_codecencodings_tw  unknown encoding
+test_codecencodings_cn
```
Comment (rth) [inline test/python_tests.txt]: Was this necessary to make CI pass with ccache? Or was it unrelated changes?

File: Makefile [hunk 2]
```diff
@@ -153,15 +154,32 @@ root/.built: \
...
+    ln -s `which ccache` $(PYODIDE_ROOT)/ccache/emcc ; \
+  else \
+    ln -s emsdk/emsdk/emscripten/tag-1.38.4/emcc $(PYODIDE_ROOT)/ccache/emcc; \
+  fi
+
+
+ccache/em++:
+	if hash ccache &>/dev/null; then \
+		mkdir -p $(PYODIDE_ROOT)/ccache ; \
+    ln -s `which ccache` $(PYODIDE_ROOT)/ccache/em++ ; \
```
Comment (rth) [inline Makefile]: maybe also set a smaller max cache size limit? e.g.
```
ccache -M 512M
```
by default it's 5 GB.

File: .circleci/config.yml [hunk 3]
```diff
@@ -58,7 +58,8 @@ jobs:
       - save_cache:
```
Comment (rth) [inline .circleci/config.yml]: Maybe also run,
```
ccache -s
```
after the `make`  to have some debugging information about ccache stats (e.g. to check that it's working as expected)?

File: Makefile (hunk 2 above)
Comment (mdboom) [reply Makefile]: Why?  I can't find what the maximum cache size is on Circle-CI, but I see no reason not to use the maximum we have available to us.

File: .circleci/config.yml (hunk 3 above)
Comment (mdboom) [reply .circleci/config.yml]: Not a bad idea.

**Comment by mdboom** [comment]
@rth: Thanks for the review. To answer your question: The codec-related changes was accidental -- I had based this branch off of the wrong one.  That should be removed now.  

Still waiting on builds... :)

File: Makefile (hunk 2 above)
Comment (rth) [reply Makefile]: I guess if it's large it still take some time to restore and it probably means that we are keeping outdated files in the cache. [Scipy does this](https://github.com/iodide-project/pyodide/pull/88#issuecomment-407480885). But you are right, we could always add it later if it ever becomes a problem.

**Pull request merged**